openhands/llm/llm.py

```diff
@@ -219,6 +219,20 @@ def wrapper(*args, **kwargs):
                         )
                     resp.choices[0].message = fn_call_response_message
 
+                message_back: str = resp['choices'][0]['message']['content'] or ''
+                tool_calls = resp['choices'][0]['message'].get('tool_calls', [])
+                if tool_calls:
+                    for tool_call in tool_calls:
+                        fn_name = tool_call.function.name
+                        fn_args = tool_call.function.arguments
+                        message_back += f'\nFunction call: {fn_name}({fn_args})'
+
+                # log the LLM response
+                self.log_response(message_back)
+
+                # post-process the response first to calculate cost
+                self._post_completion(resp)
```

Comment:
@openhands-agent Can we modify self._post_completion to return the cost, so we can directly use the returned cost for logging?